### PR TITLE
Issue 3904 - Signup with SSO

### LINKED ISSUE #3904: SSO Sign-up Page
### Description
Updated sign-up page to allow for log in with Microsoft SSO credentials.  In the main signup page there should be a link to sign up with Microsoft.
The steps are 
- Main signup page link to microsoft signup. [here](https://www.figma.com/file/3FjNEOdQbcHwzquVZxUqMl/3D-Repo?node-id=1624%3A13680&t=X24Sdk65JXqdTBqA-0)
- Then fill in the  necessary data (username/company/country) [here](https://www.figma.com/file/3FjNEOdQbcHwzquVZxUqMl/3D-Repo?node-id=1624%3A13822&t=X24Sdk65JXqdTBqA-0)
- After accepting terms and conditions [here](https://www.figma.com/file/3FjNEOdQbcHwzquVZxUqMl/3D-Repo?node-id=1083%3A10009&t=X24Sdk65JXqdTBqA-0) redirect to the microsoft sso page [here](https://www.figma.com/file/3FjNEOdQbcHwzquVZxUqMl/3D-Repo?node-id=1083%3A10059&t=X24Sdk65JXqdTBqA-0)
- If the user with sso already has an account redirect [here](https://www.figma.com/file/3FjNEOdQbcHwzquVZxUqMl/3D-Repo?node-id=1624%3A14079&t=X24Sdk65JXqdTBqA-0)

### Goals
- [x] As a user I want to sign up using microsoft single sign on
- [x] As a user if I cancel the microsoft authentication I expect to be returned to the login page
- [x] As a user I need to populate some 3D Repo specific fields (eg. username) before I am prompted for the microsoft authentication
- [x] As a user I must accept the terms and conditions before my account is created

### Tasks
- [x] Add link for microsoft sso in main signup page
- [x] Add a route for the minimum necessary data form
- [x] Add the component for the min data form
- [x] Add a route when there were already an account  in 3drepo with SSO
- [x] Add the component for showing the message that the user has already a SSO account
- [x] Apply new design

### Related Resources
[Link to design](https://www.figma.com/file/3FjNEOdQbcHwzquVZxUqMl/3D-Repo?node-id=1597%3A14060&t=aIZ6uRFkJnXtuwsP-1)

### PR DESCRIPTION
**Pull request by sanmont3drepo**
This fixes #3904 

#### Description
- Supports SSO signup with Microsoft
- Some design changes in the singup pages (logo on the left in login pages)

#### Test cases
- Try signing up with sso and  a already existing username
- Try signing up with sso and a already existing email
- Try signing up with sso with a non existing username/email

### CODE REVIEW

File: frontend/src/v5/ui/routes/userSignup/userSignupSidebar/userSignupSidebar.component.tsx [hunk 1]
```diff
@@ -16,98 +16,84 @@
...
-		<BookADemoButton
-			to={{ pathname: 'https://3drepo.com/demo/' }}
-			startIcon={<LaptopIcon />}
-		>
-			<FormattedMessage id="userSignup.sidebar.bookADemo" defaultMessage="Book A Demo" />
-		</BookADemoButton>
+		<SignUpWithMicrosoftText>
+			<FormattedMessage
+				id="userSignup.sidebar.signUpWithMicrosoft"
+				defaultMessage="Sign up with microsoft"
```
Comment (Amantini1997) [inline frontend/src/v5/ui/routes/userSignup/userSignupSidebar/userSignupSidebar.component.tsx:46]: I think Microsoft should be capitlised

**Review by Amantini1997 — CHANGES_REQUESTED** [review]

**Comment by Amantini1997** [comment]
When I try to signup, if I am already logged in with an SSO, I get logged in to that signed in user and I don't get the chance to sign up

**Comment by sanmont3drepo** [comment]
> When I try to signup, if I am already logged in with an SSO (userSSO), I get logged in as userSSO and I don't get the chance to sign up. I am not sure what is the expected process here, but I don't seem to see this image

This is when you already have a account with 3drepo (same email) but is not SSO.

**Comment by carmenfan** [comment]
> > When I try to signup, if I am already logged in with an SSO (userSSO), I get logged in as userSSO and I don't get the chance to sign up. I am not sure what is the expected process here, but I don't seem to see this image
> 
> This is when you already have a account with 3drepo (same email) but is not SSO. Maybe it should do the same with an existing username right now it says that the username has already been taken).
> 
> What do you think @carmenfan ?

yes agree

File: frontend/src/v5/ui/routes/userSignup/userSignupSidebar/userSignupSidebar.styles.ts [hunk 2]
```diff
@@ -15,92 +15,102 @@
...
 `;
 
-export const LaptopIcon = styled(LatopIconBase)`
-	width: 16px;
-	height: 16px;
+export const SidebarContent = styled.div`
+	margin-left: 4px;
+`;
+
+export const NewSticker = styled.div`
```
[the remark this answers is not in the document]
Comment (sanmont3drepo) [reply frontend/src/v5/ui/routes/userSignup/userSignupSidebar/userSignupSidebar.styles.ts:104]: TBH is not, the border radius changes, the border width changes, the padding changes. If I use the chip in the end would be the same, wont be able to reuse any of the styles.
Also this label is supossed to be removed in the in the following deploy after the next.

File: frontend/src/v5/ui/routes/login/loginSSO.component.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+export const LoginSSO = () => {
+	const { search } = useLocation();
+	const searchParams = new URLSearchParams(search);
+
+	if (searchParams.get('loginPost')) {
+		const { origin } = new URL(window.location.href);
+		window.location.href = origin;
+	}
+
+	if (!searchParams.get('loginPost')) {
```
[the remark this answers is not in the document]
Comment (sanmont3drepo) [reply frontend/src/v5/ui/routes/login/loginSSO.component.tsx:29]: That creates a lint error
image

**Comment by george-hadfield** [comment]
### Can the error message be displayed?
After trying to signup using SSO and providing an existing username, there is no error message, just the Username text and icon turns red.

image

**Comment by george-hadfield** [comment]
### The width of the box is different to design
Also the spacing between the elements if off. Worth checking against the design again

image

**Comment by george-hadfield** [comment]
### Incorrect font colour and weight

Currently using base.main but should be secondary.main. The font is one weight too heavy also

image

**Comment by george-hadfield** [comment]
### Next step button incorrectly positioned

image


The margin should be bigger below the button not above

**Comment by george-hadfield** [comment]
### Change font weight of accordion headers to 400 (currently 600)

image

**Comment by george-hadfield** [comment]
### Can't sign up using SSO

[screen-capture (5).webm](https://user-images.githubusercontent.com/78047794/222104956-8ce6c02b-8f73-489b-9817-05984523df47.webm)

**Review by george-hadfield — CHANGES_REQUESTED** [review]
Some initial changes before I can review again

File: frontend/src/v5/ui/routes/userSignup/userSignupSidebar/userSignupSidebar.styles.ts [hunk 4]
```diff
@@ -15,92 +15,102 @@
...
+	margin: 0 0 43px;
+	padding:20px;
 `;
 
-export const Tick = styled(TickBase)`
-	width: 13px;
-	height: 13px;
-	color: ${({ theme }) => theme.palette.primary.main};
+export const SignUpWithMicrosoftText = styled(Typography)`
+	${({ theme }) => theme.typography.h1};
```
[the remark this answers is not in the document]
Comment (sanmont3drepo) [reply frontend/src/v5/ui/routes/userSignup/userSignupSidebar/userSignupSidebar.styles.ts:66]: The problem is that you cant be consistent with that. I mean, we have "variants" that are not supported by Typography. In any case would make a Typography or similar component and add all variants. Maybe in this case would change the component style to div, as having it as Typography adds nothing.

**Comment by sanmont3drepo** [comment]
> ### Can the error message be displayed?
> After trying to signup using SSO and providing an existing username, there is no error message, just the Username text and icon turns red.
> 
> image

This was discussed here; to keep consistency
https://github.com/3drepo/3drepo.io/pull/4016#issuecomment-1448262535

**Comment by sanmont3drepo** [comment]
> ### Change font weight of accordion headers to 400 (currently 600)
> image

Are you sure? in figma says 500

**Comment by Amantini1997** [comment]
Previously, it was possible to move across the completed steps by clicking on the step title, now it seems only the first step is reachable this way 

https://user-images.githubusercontent.com/43547703/222217654-7d5d20be-b446-4274-988d-2896e27f8084.mp4

**Comment by Amantini1997** [comment]
When you already have an account signed up, you get this 
image

Should I not get to that page right after clicking this button (as opposed to first fill in the form and then trying to sign up)?
image

**Comment by Amantini1997** [comment]
(in incognito mode)
I tried to create an account using an already existing username and I still get redirected to the Microsoft sso page. Should I not be intercepted before reaching that far and get the error about the username already existing?

**Comment by sanmont3drepo** [comment]
> When you already have an account signed up, you get this image
> 
> Should I not get to that page right after clicking this button (as opposed to first fill in the form and then trying to sign up)? image

Nope, you need to go through the SSO process to get that one, you cant know first hand, it depends on the email and username that you filled in

**Comment by sanmont3drepo** [comment]
> (in incognito mode) I tried to create an account using an already existing username and I still get redirected to the Microsoft sso page. Should I not be intercepted before reaching that far and get the error about the username already existing?

That is discussed here
https://github.com/3drepo/3drepo.io/pull/4016#issuecomment-1448262535

**Review by Amantini1997 — APPROVED** [review]

**Comment by george-hadfield** [comment]
### Change body margin-botton to 15px

Currently set at 5px this needs to be 15px.

image

**Review by george-hadfield — CHANGES_REQUESTED** [review]
2 changes

**Review by Amantini1997 — APPROVED** [review]

**Review by george-hadfield — APPROVED** [review]

**Pull request merged**